awscli/customizations/configure/set.py

```diff
@@ -0,0 +1,104 @@
+# Copyright 2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+import os
+
+from awscli.customizations.commands import BasicCommand
+from awscli.customizations.configure.writer import ConfigFileWriter
+
+from . import PREDEFINED_SECTION_NAMES
+
+
+class ConfigureSetCommand(BasicCommand):
+    NAME = 'set'
+    DESCRIPTION = BasicCommand.FROM_FILE('configure', 'set',
+                                         '_description.rst')
+    SYNOPSIS = 'aws configure set varname value [--profile profile-name]'
+    EXAMPLES = BasicCommand.FROM_FILE('configure', 'set', '_examples.rst')
+    ARG_TABLE = [
+        {'name': 'varname',
+         'help_text': 'The name of the config value to set.',
+         'action': 'store',
+         'cli_type_name': 'string', 'positional_arg': True},
+        {'name': 'value',
+         'help_text': 'The value to set.',
+         'action': 'store',
```

Comment:
You will probably have merge conflict. Please backport this bugfix https://github.com/aws/aws-cli/pull/1791